core/src/cluster_slot_state_verifier.rs

```diff
@@ -5,12 +5,21 @@ use crate::{
 };
 use solana_ledger::blockstore::Blockstore;
 use solana_sdk::{clock::Slot, hash::Hash};
-use std::collections::{BTreeMap, BTreeSet, HashSet};
+use std::collections::{BTreeMap, BTreeSet, HashMap};
 
 pub(crate) type DuplicateSlotsTracker = BTreeSet<Slot>;
-pub(crate) type DuplicateSlotsToRepair = HashSet<(Slot, Hash)>;
+pub(crate) type DuplicateSlotsToRepair = HashMap<Slot, Hash>;
+pub(crate) type EpochSlotsFrozenSlots = BTreeMap<Slot, Hash>;
 pub(crate) type GossipDuplicateConfirmedSlots = BTreeMap<Slot, Hash>;
 
+#[derive(PartialEq, Clone, Debug)]
+pub enum ClusterConfirmedHash {
```

Comment:
Introduce `ClusterConfirmedHash` to track what level of certainty we have about a slot